controllers/userController.js

```diff
@@ -7,23 +11,52 @@ const getUsers = (req, res) => {
     res.json(users)
 };
 
-const createUser = (req, res) => {
-    const name = req.body.name;
-    const email = req.body.email;
-    const user = {
+const createUser = async (req, res) => {
+    try {
+        const salt = await bcrypt.genSalt();
+        const hashPassword = await bcrypt.hash(req.body.password, salt);
+        const name = req.body.name;
+        const email = req.body.email;
+        const password = hashPassword;
+        const user = {
         id: uuidv4(),
         email,
         name,
+        password
     };
-    if (!user.name) {return res.status(400).send('A name should be provided')}
+    if (!user.name) {res.status(400).send('A name should be provided')}
     users.push(user);
-    res.status(201).json(user);
+    res.status(201);
+
+    // Send JSON WEB TOKEN
+    const token = JWT.sign(user, process.env.SECRET_KEY)
+    res.json({token})
+    } catch {
+        res.status(500).sendStatus(500);
+    }
 };
 
+const loginUser = async (req, res) => {
+    const name = req.body.name;
+    const findUser = users.find((user) => user.name === name);
+    if(!findUser) {res.status(404).send('The User with this NAME does not exist')};
+    try {
+        if (await bcrypt.compare(req.body.password, findUser.password))
+        {
+            const token = JWT.sign(findUser, process.env.SECRET_KEY)
+            res.json({token})
+        } else {
+            res.send('Password does not match')
```

Comment:
Looks like you are mixing and matching string and json responses. Typically, you stick to one. In this case, please use JSON. here is an example of the format I think you should use for all response body

```json
{
    statusCode: 401,
    message: "Password does not match our records.",
}
```

Reply:
noted, will work on this